core/src/test/scala/org/apache/spark/deploy/security/KafkaTokenUtilSuite.scala

```diff
@@ -78,6 +79,19 @@ class KafkaTokenUtilSuite extends SparkFunSuite with BeforeAndAfterEach {
     Configuration.setConfiguration(null)
   }
 
+  test("checkProxyUser with proxy current user should throw exception") {
+    val currentUser = UserGroupInformation.getCurrentUser()
+    val savedAuthenticationMethod = currentUser.getAuthenticationMethod()
+    try {
+      currentUser.setAuthenticationMethod(UserGroupInformation.AuthenticationMethod.PROXY)
+      intercept[IllegalArgumentException] {
+        KafkaTokenUtil.checkProxyUser()
+      }
```

Comment:
Please check the explicit error message. We need to make it sure that the correct `IllegalArgumentException` is thrown since there are many `IllegalArgumentException` cases.

Reply:
Yeah, it makes it more secure. Fixed.